### DISCUSSION TITLE
Generating a 24-hour LDFC Spectrogram from Intermittent Short Clips

### DISCUSSION THREAD

**Discussion by mtshiao** [QutEcoacoustics/audio-analysis#559] [Q&A]
Hello,

I'm a beginner using AnalysisPrograms and I'm particularly interested in the audio2csv command to convert audio files into false-colour long-duration spectrograms. From the documentation and related literature, it seems this command is designed for 24-hour long recordings. However, my recording device is set up to record for 1 minute every 3 minutes (with a 2-minute pause), resulting in 480 one-minute clips throughout the day.

Is it possible for me to use this program to generate a 24-hour LDFC spectrogram from these short clips? Should I be concatenating the spectrograms generated from each one-minute clip horizontally? If I do this, the resulting graph does not have any X and Y axis ticks. Or is there a parameter I can modify directly to achieve this? 

I would appreciate any guidance or suggestions.

Thank you in advance for your help.

**Comment by atruskie** [comment]
hiya!

Good question. 

The basic process is to generate the indices for each input file, and the use the concatenate function to draw the combined image.

Here is a powershell script that demonstrates the process: https://github.com/QutEcoacoustics/audio-analysis/blob/master/scripts/indices_and_concat.ps1 (PowerShell may not be your language of choice but it should be easy enough to adapt the code to your desired language).

You'll want to pay particular attention to the gap rendering option because of your sampling regime. From `AP concatenateindexfiles --help`:

```
  -g|--gap-rendering <GAP_RENDERING>     How to render gaps in a recording. Valid options: `TimedGaps`
                                                 (default), `NoGaps`, `EchoGaps`
                                                 Allowed values are: TimedGaps, NoGaps, EchoGaps
```

https://ap.qut.ecoacoustics.info/technical/api/audioanalysistools.indices.concatmode

That should be enough to get you started. Let me know if you need more help.

**Reply by mtshiao** [reply]
Hi, atruskie

Thanks so much for your previous tips - they were super helpful! Following your advice and the resources available on GitHub, I successfully managed to concatenate multiple audio file spectrograms. However, after completing the process, I noticed there's still a small gray segment appearing in the high-frequency section of the merged spectrogram.  Here's the code I ran in R:

```
cmd_concat <- sprintf('"%s" ConcatenateIndexFiles --input-data-directory "%s/indices" --output-directory "%s" -z %s --file-stem-name %s --directory-filter "*.*" --index-properties-config "%s/ConfigFiles/IndexPropertiesConfig.yml" --false-colour-spectrogram-config "%s/ConfigFiles/SpectrogramFalseColourConfig.yml" --draw-images --no-debug --gap-rendering EchoGaps', ap_path, output_directory, output_directory, time_zone_offset, basename(input_directory), default_configs, default_configs)
system(cmd_concat)
```

I'm curious if you might have any insights on why this gray segment appears and how it could potentially be resolved.

[EchoGaps]
![FSaudio_20230219__2Maps_EchoGaps](https://github.com/QutEcoacoustics/audio-analysis/assets/118085797/e5b9cf2d-ffc4-41ac-8e5e-a387aaddc1b7)

[NoGaps]
![FSaudio_20230219__2Maps_NoGap](https://github.com/QutEcoacoustics/audio-analysis/assets/118085797/f74bc267-294c-4d0f-8502-9224c1ccbbc6)


Once again, thank you for your assistance thus far.

**Comment by atruskie** [comment]
Nice work, the echo gaps one looks pretty serviceable!

There was a reason for this, it's some kind of marker and I think it shows the start of each new segment... (With the no gaps one, the marker is so frequent that it looks like a solid bar). I'll see if I can hunt it down... 

Yup here it is: https://github.com/QutEcoacoustics/audio-analysis/blob/23a0bbf05c5f59638dbb195a7bc613096d64a6b5/src/AudioAnalysisTools/Indices/GapsAndJoins.cs#L512

It's rendering spots where we've found some kind of error when combining the data. There should be a gaps and joins file somewhere I the output that explains the nature of the supposed errors.

**Reply by mtshiao** [reply]
Thank you once again for your detailed response. Your insights have provided me with a deeper understanding of the results. I truly appreciate your assistance!
